### DISCUSSION TITLE
Unstacking and organizing a stack of images

### DISCUSSION THREAD

**Discussion by brownphotographic** [photoprism/photoprism#4514] [Ideas]
As a user I'd like to be able to open up a stack of images temporarily to see all images in the stack.

e.g. if a stack has a RAW + JPEG image, I'd like to be able to get to either of the files easily. Currently I can only see the RAW, which appears to be the stack 'pick'

Could we investigate a solution like Lightroom, Apple Aperture, or Immich where a stack of images can:
a) be clearly seen by an icon on the thumbnail
b) expanded by clicking on that thumbnail. When the stack is expanded, show which images belong to that stack (vs other images around the stack). Aperture did this best.
c) reorder the stack to decide which one should be at the top of the stack when the stack is closed. e.g. drag the pick to the left, or select 'stack pick'

I'm not sure if PhotoPrism supports manually stacking images yet (like Immich does for example). If it did, then the above could apply.

**Comment by graciousgrey** [comment]
Thank you for your feedback!

You can view all files that belong to a photo on the files tab of the photo edit dialog. There you can unstack files or change the primary file of a stack. Manual stacking is not possible yet but planned for the future.

You can find more information on stacks here: https://docs.photoprism.app/user-guide/organize/stacks/

**Reply by brownphotographic** [reply]
Ok thank you

I see this now thanks for the tip

It would be great to see images that have been associated with one another to be represented as a stack wherever they are accessed. 

This is something that Lightroom does well.

Use case... if I edit an image I may make several versions of that same image. Eg one is the raw file, another is a black and white render, another a color render. They are all associated with each other, but all are valid and therefore to the user they want them to be easily accessible.

I guess there are two paradigms here for PhotoPrism
1. Where photoprism is a simple image viewer. The user does there editing and curation outside of the tool and just exports images, then imports to the tool.
2. Where photoprism is used to manage and navigate a photo library and point the tool at the photo repository. 

I'm trialing PhotoPrism to do number 2, and it may not be built for this. Perhaps such a feature could help all users regardless of use case but of course it shouldn't solve all the problems. 

Thanks for listening!
